packages/iframe-execution-environment-service/src/IframeExecutionEnvironmentService.ts

```diff
@@ -0,0 +1,341 @@
+
+    const jobWrapper = this.jobs.get(jobId);
+    if (!jobWrapper) {
+      throw new Error(`Job with id "${jobId}" not found.`);
+    }
+
+    console.log('Parent: Sending Command', message);
+    const response: PendingJsonRpcResponse<unknown> =
+      await jobWrapper.rpcEngine.handle(message);
+    if (response.error) {
+      throw new Error(response.error.message);
+    }
+    return response.result;
+  }
+
+  async terminateAllPlugins() {
+    for (const jobId of this.jobs.keys()) {
+      this.terminate(jobId);
+    }
+    this._pluginRpcHooks.clear();
+  }
+
+  async terminatePlugin(pluginName: string) {
+    const jobId = this.pluginToJobMap.get(pluginName);
+    jobId && this.terminate(jobId);
+    this._removePluginHooks(pluginName);
+  }
+
+  terminate(jobId: string): void {
+    const jobWrapper = this.jobs.get(jobId);
+    if (!jobWrapper) {
+      throw new Error(`Job with id "${jobId}" not found.`);
+    }
+
+    Object.values(jobWrapper.streams).forEach((stream) => {
+      try {
+        !stream.destroyed && stream.destroy();
+        stream.removeAllListeners();
+      } catch (err) {
+        console.log('Error while destroying stream', err);
+      }
+    });
+    document.getElementById(jobWrapper.id)?.remove();
+    this._removePluginAndJobMapping(jobId);
+    this._deleteJob(jobId);
+    console.log(`job: "${jobId}" terminated`);
+  }
+
+  /**
+   * Gets the RPC message handler for the given plugin.
+   *
+   * @param pluginName - The name of the plugin whose message handler to get.
+   */
+  async getRpcMessageHandler(pluginName: string) {
+    return this._pluginRpcHooks.get(pluginName);
+  }
+
+  private _removePluginHooks(pluginName: string) {
+    this._pluginRpcHooks.delete(pluginName);
+  }
+
+  private _createPluginHooks(pluginName: string, jobId: string) {
+    const rpcHook = async (
+      origin: string,
+      request: Record<string, unknown>,
+    ) => {
+      return await this._command(jobId, {
+        id: nanoid(),
+        jsonrpc: '2.0',
+        method: 'pluginRpc',
+        params: {
+          origin,
+          request,
+          target: pluginName,
+        },
+      });
+    };
+
+    this._pluginRpcHooks.set(pluginName, rpcHook);
+  }
+
+  async executePlugin(pluginData: PluginData): Promise<unknown> {
+    if (this.pluginToJobMap.has(pluginData.pluginName)) {
+      throw new Error(
+        `Plugin "${pluginData.pluginName}" is already being executed.`,
+      );
+    }
+
+    const job = await this._init();
+    this._mapPluginAndJob(pluginData.pluginName, job.id);
+    this.setupPluginProvider(
+      pluginData.pluginName,
+      job.streams.rpc as unknown as Duplex,
+    );
+
+    const result = await this._command(job.id, {
+      jsonrpc: '2.0',
+      method: 'executePlugin',
+      params: pluginData,
+      id: nanoid(),
+    });
+    this._createPluginHooks(pluginData.pluginName, job.id);
+    return result;
+  }
+
+  _mapPluginAndJob(pluginName: string, jobId: string): void {
```

Comment:
This is also an old hangover, but a lot / all of these `_`-prefixed methods should be `private`.